requirements.txt

```diff
@@ -1 +1,12 @@
+pip
 urllib3[secure]>=1.23
+coverage
+codecov
+flake8
+pylint
+fastimport
+mypy
+setuptools
+wheel
+twine
+fastimport
```

Comment:
In addition, this part may need to be considered by you. After all, most of the new additions are used to check the Python code, not the dependent packages required by the Dulwich. I can revert these changes when I submit again.

Reply:
None of the new additions are strict requirements for Dulwich. fastimport is an optional but recommended dependency, so I think you could argue that that should be here.

The other libraries are for specific use situations only (uploading to pip, linting, etc) and should not need to be installed by regular users of dulwich.

Reply:
@jelmer Understand, I will modify this part according to your suggestions.